app/objects/c_adversary.py

```diff
@@ -71,6 +80,20 @@ def store(self, ram):
         existing.update('has_repeatable_abilities', self.check_repeatable_abilities(ram['abilities']))
         return existing
 
+    def verify(self, log, abilities, objectives):
```

Comment:
This is called `verify` but it has no return and has some logged checks and populates the default objectives when they are missing and fills out the repeatable abilities flag.  I am fine with leaving it since the PR is large (and it seems the calling function/refactored code did the same), but it seems to be named different from its real function.